src/translate.py

```diff
@@ -170,52 +170,54 @@ def indent(self, indent: int, line: str) -> str:
         return self.indent_step * max(indent + self.extra_indent, 0) + line
 
 
-def as_type(expr: "Expression", type: Type, silent: bool) -> "Expression":
-    if expr.type.unify(type):
+def as_type(
+    expr: "Expression", type: Type, typemap: Optional[TypeMap], silent: bool
+) -> "Expression":
+    if expr.type.unify(type, typemap):
         if not silent and not isinstance(expr, Literal):
             return Cast(expr=expr, reinterpret=True, silent=False, type=type)
         return expr
     return Cast(expr=expr, reinterpret=True, silent=False, type=type)
 
 
 def as_f32(expr: "Expression") -> "Expression":
-    return as_type(expr, Type.f32(), True)
+    return as_type(expr, Type.f32(), None, True)
```

Comment:
having to pass None here without a lot more messy plumbing seems like a downside of this approach. It may well be fine in practice, especially since we'll unify on use(), but passing typemaps consistently would be nicer